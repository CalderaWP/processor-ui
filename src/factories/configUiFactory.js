import * as CalderaComponents from '@caldera-labs/components';

/**
 * Given an array of field configurations, return array of React components for them
 *
 *
 * @param {Array}fieldConfigs
 * @return {Array<React.Component>}
 */
export const configUiFactory = (fieldConfigs) => {
	return CalderaComponents.factories.fieldSetFactory(fieldConfigs);
};
