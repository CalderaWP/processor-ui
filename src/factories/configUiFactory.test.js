import renderer from 'react-test-renderer';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import {configUiFactory} from "./configUiFactory";

describe( 'configUiFactory',  () => {

	function change(){}
	const fieldsSet = [];
	//A mock library is needed!
	const textFieldConfig = {
		id: 'tags',
		'label': 'Tags',
		'description': 'Comma separated list of tags.',
		'type': 'text',
		'default': false,
		onValueChange: change,
	};

	const defaultSequenceId = 42;
	const hiddenFieldConfig = {
		'id': 'seq',
		'type': 'hidden',
		'label': 'Sequence ID',
		'default': defaultSequenceId,

		onValueChange: change,
	};

	const fields = {
		tags: textFieldConfig,
		sequenceId: hiddenFieldConfig
	};

	Object.values( fields ).map( field =>{
		fieldsSet.push(field)
	});

	it( 'matches the snapshot', () =>{
		const component = renderer.create(
			<div className={'test-group'}>
				{Array.from(configUiFactory(fields)).map((field,i) => {
					return React.createElement(
						'div', {
							key: i,
							className: `f-${i}`
						},
						field
					);
				})}
			</div>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});


	it( 'Renders the right UI', () =>{
		const wrapper = mount(
			<div className={'test-group'}>
				{Array.from(configUiFactory(fields)).map((field,i) => {
					return React.createElement(
						'div', {
							key: i,
							className: `f-${i}`
						},
						field
					);
				})}
			</div>
		);
		expect( wrapper.find('.f-1').children().length ).toEqual(1);
		expect( wrapper.find('.test-group').children().length ).toEqual(2);
	});



});