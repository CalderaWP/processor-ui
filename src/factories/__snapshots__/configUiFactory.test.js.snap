// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configUiFactory matches the snapshot 1`] = `
<div
  className="test-group"
>
  <div
    className="f-0"
  >
    <div
      className="caldera-config-group"
    >
      <label
        htmlFor="tags"
      >
        Tags
      </label>
      <input
        aria-describedby={null}
        className="field-config"
        id="tags"
        onChange={[Function]}
        required={false}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="f-1"
  >
    <div
      className="caldera-config-group"
    >
      <label
        htmlFor="seq"
      >
        Sequence ID
      </label>
      <input
        aria-describedby={null}
        className="field-config"
        id="seq"
        onChange={[Function]}
        required={false}
        type="hidden"
        value={undefined}
      />
    </div>
  </div>
</div>
`;
